src/components/zen-table/zen-table-helpers.tsx

```diff
@@ -64,22 +78,34 @@ export const cleanupTableStructure = function (table: HTMLZenTableElement): void
     }
   };
 
+  const updateHeaderSelectCheckbox = (header, rows) => {
+    const allSelected = rows.length && rows.every(n => n.selected);
+    const someSelected = rows.some(n => n.selected);
```

Comment:
```suggestion
    const someSelected = !allSelected && rows.some(n => n.selected);
```

There's no need to do the `.some` if the `.every` returned `true`.